docs/advanced/customization-guide.md

```diff
@@ -16,8 +15,1007 @@ published: false
 
 ---
 
-***WORK IN PROGRESS.***
+## Overview
+
+NFD provides multiple extension points for vendor and application specific
+labeling:
+
+- [NodeFeatureRule](#nodefeaturerule-custom-resource) objects provide a way to
+  deploy custom labeling rules via the Kubernetes API
+- [*local* feature source](#local-feature-source) of nfd-worker creates
+  labels by executing hooks and reading files
+- [*custom* feature source](#custom-feature-source) of nfd-worker creates
+  labels based on user-specified rules
+
+## NodeFeatureRule custom resource
+
+NodeFeatureRule objects provide an easy way to create vendor or application
+specific labels. It uses a flexible rule-based mechanism for creating labels
+based on node feature.
+
+### A NodeFeatureRule example
+
+Consider the following referential example:
+
+```yaml
+apiVersion: nfd.k8s-sigs.io/v1alpha1
+kind: NodeFeatureRule
+metadata:
+  name: my-rule-object
+spec:
+  rules:
+    - name: "my sgx feature rule"
+      labels:
+        "my-sgx-feature": "true"
+      matchFeatures:
+        - feature: cpu.cpuid
+          matchExpressions:
+            SGX: {op: Exists}
+        - feature: kernel.config
+          matchExpressions:
+            X86_SGX: {op: In, value: ["y"]}
+```
+
+It specifies one rule which creates node label
+`feature.node.kubenernetes.io/my-sgx-feature=true` if both of the following
+conditions are true (`matchFeatures` implements a logical AND over the
+matchers):
+
+- the CPU has SGX capability
+- X86_SGX option has been enabled in the kernel
+
```

Comment:
maybe add the basic `$ kubectl apply -f nfd-rule-sgx.yaml` to show how these are deployed, or maybe also the syntax how CRs can be added via `kustomization.yaml`.

Reply:
Makes sense. Maybe we could even include the sample CR in the repo 🧐

Reply:
Added a kubectl example and added a sample CR yaml in the repo